libraries/chain/include/graphene/chain/betting_market_object.hpp

```diff
@@ -0,0 +1,716 @@
+          return true;
+      if (lhs_bettor_id > rhs_bettor_id)
+          return false;
+      if (lhs_betting_market_id < rhs_betting_market_id)
+         return true;
+      if (lhs_betting_market_id > rhs_betting_market_id)
+         return false;
+      return lhs_bet_type < rhs_bet_type;
+   }
+   bool compare(const account_id_type& lhs_bettor_id, const betting_market_id_type& lhs_betting_market_id,
+                bet_type lhs_bet_type, 
+                bet_multiplier_type lhs_backer_multiplier,
+                const account_id_type& rhs_bettor_id, const betting_market_id_type& rhs_betting_market_id, 
+                bet_type rhs_bet_type,
+                bet_multiplier_type rhs_backer_multiplier) const
+   {
+      if (lhs_bettor_id < rhs_bettor_id)
+          return true;
+      if (lhs_bettor_id > rhs_bettor_id)
+          return false;
+      if (lhs_betting_market_id < rhs_betting_market_id)
+         return true;
+      if (lhs_betting_market_id > rhs_betting_market_id)
+         return false;
+      if (lhs_bet_type < rhs_bet_type)
+         return true;
+      if (lhs_bet_type > rhs_bet_type)
+         return false;
+      if (lhs_bet_type == bet_type::back)
+        return lhs_backer_multiplier < rhs_backer_multiplier;
+      else
+        return lhs_backer_multiplier > rhs_backer_multiplier;
+   }
+   bool compare(const account_id_type& lhs_bettor_id, const betting_market_id_type& lhs_betting_market_id, 
+                bet_type lhs_bet_type, 
+                bet_multiplier_type lhs_backer_multiplier, const bet_id_type& lhs_bet_id,
+                const account_id_type& rhs_bettor_id, const betting_market_id_type& rhs_betting_market_id,
+                bet_type rhs_bet_type,
+                bet_multiplier_type rhs_backer_multiplier, const bet_id_type& rhs_bet_id) const
+   {
+      if (lhs_bettor_id < rhs_bettor_id)
+          return true;
+      if (lhs_bettor_id > rhs_bettor_id)
+          return false;
+      if (lhs_betting_market_id < rhs_betting_market_id)
+         return true;
+      if (lhs_betting_market_id > rhs_betting_market_id)
+         return false;
+      if (lhs_bet_type < rhs_bet_type)
+         return true;
+      if (lhs_bet_type > rhs_bet_type)
+         return false;
+      if (lhs_backer_multiplier < rhs_backer_multiplier)
+         return lhs_bet_type == bet_type::back;
+      if (lhs_backer_multiplier > rhs_backer_multiplier)
+         return lhs_bet_type == bet_type::lay;
+      
+      return lhs_bet_id < rhs_bet_id;
+   }
+};
+
+struct by_odds {};
+struct by_betting_market_id {};
+struct by_bettor_and_odds {};
+typedef multi_index_container<
+   bet_object,
+   indexed_by<
+      ordered_unique< tag<by_id>, member< object, object_id_type, &object::id > >,
+      ordered_unique< tag<by_odds>, identity<bet_object>, compare_bet_by_odds >,
+      ordered_non_unique< tag<by_betting_market_id>, member<bet_object, betting_market_id_type, &bet_object::betting_market_id> >,
```

Comment:
This should be made `ordered_unique` to enforce a well-defined order when iterating.